src/wallet/index.ts

```diff
@@ -23,12 +23,27 @@ import { SignOptions } from "./transaction/types";
 class Wallet {
   readonly publicKey: string;
   readonly privateKey: string;
+  readonly classicAddress: string;
+  readonly seed?: string;
   private static readonly defaultAlgorithm: ECDSA = ECDSA.ed25519;
   private static readonly defaultDerivationPath: string = "m/44'/144'/0'/0/0";
 
-  constructor(publicKey: string, privateKey: string) {
+  constructor(publicKey: string, privateKey: string, seed?: string) {
```

Comment:
Yeah it does look weird but `Wallet.generate()` calls this constructor to save the new seed created with its public/private keypair.

Reply:
Like here: https://github.com/XRPLF/xrpl-py/blob/master/xrpl/wallet/main.py#L19

Reply:
This would be possible except the issue is with `Wallet.fromMnemonic()` where its seed value is in a different format than the other derive functions. It uses `bip32` instead of `ripple-keypairs` to derive public/private keypair from its seed value.